### ISSUE TITLE
Suddenly getting traceback: AttributeError: 'NoneType' object has no attribute 'items'

### ISSUE THREAD

**Issue by garretwilson** [aws/aws-sam-cli#6014]
I'm on Windows 10. A day or two ago I updated to SAM CLI, version 1.97.0. I don't know for sure if it's related to the upgrade, but now deploying a very simple template gives me:

```
Traceback (most recent call last):
  File "runpy.py", line 194, in _run_module_as_main
  File "runpy.py", line 87, in _run_code
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\samcli\__main__.py", line 12, in <module>
    cli(prog_name="sam")
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\click\core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\click\core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\click\core.py", line 1686, in invoke
    sub_ctx = cmd.make_context(cmd_name, args, parent=ctx)
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\click\core.py", line 943, in make_context
    self.parse_args(ctx, args)
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\click\core.py", line 1408, in parse_args
    value, args = param.handle_parse_result(ctx, opts, args)
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\click\core.py", line 2400, in handle_parse_result
    value = self.process_value(ctx, value)
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\click\core.py", line 2362, in process_value
    value = self.callback(ctx, self, value)
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\samcli\commands\_utils\options.py", line 170, in artifact_callback
    for _template_artifact in get_template_artifacts_format(template_file=template_file)
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\samcli\commands\_utils\template.py", line 301, in get_template_artifacts_format
    for _, resource in template_dict.get("Resources", {}).items():
AttributeError: 'NoneType' object has no attribute 'items'
```

Why?!

**Comment by garretwilson** [comment]
After a reboot the error is no longer occurring. I also remember that the error started occurring after I used a VPN and then turned off the VPN.

Is it possible that Python could be caching some sort of network configuration from the VPN? I don't have time to test the VPN again at the moment, but I wanted to leave this extra detail.

**Comment by sriram-mv** [comment]
I tried to reproduce this @garretwilson but could not. It's strange that it was transient and its no longer manifesting when turning off the VPN. It should not have any bearing on this. 🤔

**Comment by garretwilson** [comment]
It is strange. Today I tried the same VPN connection, and used SAM both with the VPN on and after turning off the VPN connection. The problem did not occur.

Thus at the moment I cannot reproduce it either.

**Comment by hnnasit** [comment]
Closing this issue as it seems like a transient issue which cannot be reproduced. Feel free to open a new issue if you still face similar issues with the latest SAM CLI version or have steps to reproduce it.

**Comment by garretwilson** [comment]
I'm suddenly getting this error again. Please re-open this issue.

(I actually searched on the error message and found my own ticket. 🤦)

No VPN was involved this time. It is a very simple template that uses `Fn::ImportValue` to access exports. I was using v1.135.0, and I upgraded to v1.145.2; I get the same error (although v1.145.2 color-codes the errors):

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\Lib\site-packages\samcli\__main__.py", line 12, in <module>
    cli(prog_name="sam")
    ~~~^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\Lib\site-packages\click\core.py", line 1161, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\Lib\site-packages\click\core.py", line 1082, in main
    rv = self.invoke(ctx)
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\Lib\site-packages\click\core.py", line 1695, in invoke
    sub_ctx = cmd.make_context(cmd_name, args, parent=ctx)
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\Lib\site-packages\click\core.py", line 949, in make_context
    self.parse_args(ctx, args)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\Lib\site-packages\click\core.py", line 1417, in parse_args
    value, args = param.handle_parse_result(ctx, opts, args)
                  ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\Lib\site-packages\click\core.py", line 2403, in handle_parse_result
    value = self.process_value(ctx, value)
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\Lib\site-packages\click\core.py", line 2365, in process_value
    value = self.callback(ctx, self, value)
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\Lib\site-packages\samcli\commands\_utils\options.py", line 196, in artifact_callback
    for _template_artifact in get_template_artifacts_format(template_file=template_file)
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\Lib\site-packages\samcli\commands\_utils\template.py", line 306, in get_template_artifacts_format
    for _, resource in template_dict.get("Resources", {}).items():
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'items'
```

I'm now on Windows 11.

**Comment by garretwilson** [comment]
Ah ha! I may have found the issue. The `template_dict.get("Resources", {}).items()` and `'NoneType' object has no attribute 'items'` seem to mean that no resources were found.

In this particular case I had copied the two resources from elsewhere, but they were not indented under the `Resources:` section, so in YAML terms they were items at the same level of `Resources`, and `Resources` had no children. Indenting the two resources resolved this issue.

I don't know if this was the source of the error when I filed this ticket, but I would still encourage you to add the minimal amount of error checking in the CLI to simply check "if resources.length() is zero, tell the user that there is nothing to deploy" or something. Just letting the program crash if the input isn't as expected doesn't make for the most user-friendly experience.

Hopefully if someone else runs into this in the future they can find help by discovering this ticket.
